warrior/Actions/CommonActions/common_actions.py

```diff
@@ -172,29 +172,34 @@ def verify_data(self, expected, object_key, type='str', comparison='eq'):
             return False
 
     def set_env_var(self, var_key=None, var_value=None, filepath=None,
-                    jsonkey="environmental_variables"):
-        """create a temp environment variable
-        the value will only stay for this run
+                    jsonkey="environmental_variables", overwrite = "yes"):
+        """Create a temp environment variable, the value will only stay for the current Execution
         :Argument:
             var_key = key of the environment variable
             var_value = value of the environment variable
             filepath = Json file where Environmental variables are defined
             jsonkey = The key where all the ENV variable & values are defined
         With jsonkey arg, Users can call same file to set various ENV Variable
+            overwrite = Yes-Will overwrite ENV variables set earlier via terminal or other means
+                        No -Will not overwrite the ENV variables set earlier with the ones passed
+                            through this keyword.
 
         Variable File :
         Sample environmental_variable file is available under
         Warriorspace/Config_file/Samples/Set_ENV_Variable_Sample.json
         """
+        overwrite = overwrite.upper()
         status = False
         if not any([var_key, var_value, filepath]):
-            print_error('Either Provide values to arguments \"var_key\" & \"var_value\" or to argument \"filepath\"')
-
-        if var_key is not None and var_value is not None:
+            print_error('Either Provide values to arguments \"var_key\" & \"var_value\" or to '
+                        'argument \"filepath\"')
+        if overwrite == "NO" and os.getenv(var_key):
+            print_info("Using ENV variable {0} set earlier with value '{1}'".format(var_key,
+                                                                           os.getenv(var_key)))
+        if var_key is not None and var_value is not None and overwrite == "YES":
```

Comment:
this condition should work for overwrite == "NO" condition as well.
Use-case - when overwrite is "NO" and the value is not set earlier.